module/input/ImageDecompressor/src/ImageDecompressor.cpp

```diff
@@ -9,130 +9,128 @@
             }
-        });
-
-        on<Trigger<CompressedImage>>().then("Decompress Image", [this](const CompressedImage& image) {
-            // Find our list of compressors, and if needed recreate it
-            std::shared_ptr<DecompressorContext> ctx;
-
-            /* Mutex Scope */ {
-                std::lock_guard<std::mutex> lock(decompressor_mutex);
-                auto it = decompressors.find(image.camera_id);
-                if (it == decompressors.end() || it->second->width != image.dimensions[0]
-                    || it->second->height != image.dimensions[1] || it->second->format != image.format) {
-                    log<NUClear::INFO>("Rebuilding decompressors for", image.name, "camera");
-
-                    // Replace the existing one with a new one
-                    it = decompressors.insert(std::make_pair(image.camera_id, std::make_shared<DecompressorContext>()))
-                             .first;
-                    it->second->width  = image.dimensions[0];
-                    it->second->height = image.dimensions[1];
-                    it->second->format = image.format;
-
-                    for (auto& f : config.factories) {
-                        for (int i = 0; i < f.second; ++i) {
-                            it->second->decompressors.emplace_back(DecompressorContext::Decompressor{
-                                std::make_unique<std::atomic<bool>>(),
-                                f.first->make_decompressor(image.dimensions[0], image.dimensions[1], image.format),
-                            });
-                        }
+        }
+    });
+
+    on<Trigger<CompressedImage>>().then("Decompress Image", [this](const CompressedImage& image) {
+        // Find our list of compressors, and if needed recreate it
+        std::shared_ptr<DecompressorContext> ctx;
+
+        /* Mutex Scope */ {
+            std::lock_guard<std::mutex> lock(decompressor_mutex);
+            auto it = decompressors.find(image.id);
+            if (it == decompressors.end() || it->second->width != image.dimensions[0]
+                || it->second->height != image.dimensions[1] || it->second->format != image.format) {
+                log<NUClear::INFO>("Rebuilding decompressors for", image.name, "camera");
+
+                // Replace the existing one with a new one
+                it = decompressors.insert(std::make_pair(image.id, std::make_shared<DecompressorContext>())).first;
+                it->second->width  = image.dimensions[0];
+                it->second->height = image.dimensions[1];
+                it->second->format = image.format;
+
+                for (auto& f : config.factories) {
+                    for (int i = 0; i < f.second; ++i) {
+                        auto a = std::make_unique<std::atomic<bool>>();
+                        it->second->decompressors.emplace_back(DecompressorContext::Decompressor{
+                            std::move(a),
```

Comment:
It seems a bit redundant to create a lvalue then immediately move it.

Reply:
Wasn't previously this way, but clang static analysis flags this as a memory leak (incorrectly) if you don't do it this way.